src/module_4/module_4_model-fitting.md

```diff
@@ -0,0 +1,1214 @@
+    Train_ce 0.06103491979394085. Val_ce 0.05640256031956773
+
+
+
+    
+![png](module_4_model-fitting_files/module_4_model-fitting_28_1.png)
+    
+
+
+    Precision Recall AUC: 0.19890975902943264
+    Roc Curve AUC: 0.844747437906795
+    Sum of both AUCs: 1.0436571969362276
+
+
+
+    
+![png](module_4_model-fitting_files/module_4_model-fitting_28_3.png)
+    
+
+
+    Precision Recall AUC: 0.19178290046990729
+    Roc Curve AUC: 0.8463536274974822
+    Sum of both AUCs: 1.0381365279673895
+
+
+
+```python
+plot_feature_importance(gbt, train_cols)
+```
+
+
+
+
+    (<Figure size 640x480 with 1 Axes>, <Axes: ylabel='features'>)
+
+
+
+
+    
+![png](module_4_model-fitting_files/module_4_model-fitting_29_1.png)
+    
+
+
+### Model 4: Gradient Boosting Three - Without categorical variables - variables with importance > 0.02
+
+Intentaremos mejorar nuestro modelo una vez más al reducir su complejidad, eliminando aquellos coeficientes que resulten menos significativos.
+
+
+```python
+train_cols = ["global_popularity", "days_since_purchase_product_type", "user_order_seq", "days_since_purchase_variant_id", "abandoned_before", "std_days_to_buy_variant_id", "avg_days_to_buy_variant_id", "set_as_regular", "discount_pct", "normalised_price", "ordered_before", "std_days_to_buy_product_type", "avg_days_to_buy_product_type"]
+
+lr = 0.05
+depth = 5
+n_trees = 100
+
+gbt = GradientBoostingClassifier(learning_rate=lr, max_depth=depth, n_estimators=n_trees)
+gbt.fit(X_train[train_cols], y_train)
+
+train_probs = gbt.predict_proba(X_train[train_cols])[:, 1]
+test_probs = gbt.predict_proba(X_test[train_cols])[:, 1]
+val_probs = gbt.predict_proba(X_val[train_cols])[:, 1]
+
+print(f"Train_auc: {roc_auc_score(y_train, train_probs)}.",f"Val_auc {roc_auc_score(y_val, val_probs)}")
+print(f"Train_ap {average_precision_score(y_train, train_probs)}.",f"Val_ap {average_precision_score(y_val, val_probs)}")
+print(f"Train_ce {log_loss(y_train, train_probs)}.",f"Val_ce {log_loss(y_val, val_probs)}")
+
+plot_metrics("Random_Forest_2", test_probs, y_test)
+plot_metrics("Random_Forest_2", val_probs, y_val)
+```
+
+    Train_auc: 0.843473214235045. Val_auc 0.846327982213392
+    Train_ap 0.21841262473539977. Val_ap 0.19257618354668404
+    Train_ce 0.06105798499900578. Val_ce 0.0563746229992203
+
+
+
+    
+![png](module_4_model-fitting_files/module_4_model-fitting_31_1.png)
+    
+
+
+    Precision Recall AUC: 0.1992181888589764
+    Roc Curve AUC: 0.8448881570818715
+    Sum of both AUCs: 1.044106345940848
+
+
+
+    
+![png](module_4_model-fitting_files/module_4_model-fitting_31_3.png)
+    
+
+
+    Precision Recall AUC: 0.19235713920249017
+    Roc Curve AUC: 0.846327982213392
+    Sum of both AUCs: 1.038685121415882
+
+
+
+```python
+plot_feature_importance(gbt, train_cols)
+```
+
+
+
+
+    (<Figure size 640x480 with 1 Axes>, <Axes: ylabel='features'>)
+
+
+
+
+    
+![png](module_4_model-fitting_files/module_4_model-fitting_32_1.png)
```

Comment:
I am missing a plot where you put all the metrics from all the different models to see how much do they change from one model to another one (is the best model just by chance or is clearly superior)

Reply:
Also, at the end you should take the best model and evaluate it in test (which you are not supposed to touch until now)